docs/apidiffs/current_vs_latest/opentelemetry-sdk.txt

```diff
@@ -1,2 +1,9 @@
 Comparing source compatibility of  against 
-No changes.
\ No newline at end of file
+***! MODIFIED CLASS: PUBLIC FINAL io.opentelemetry.sdk.OpenTelemetrySdk  (not serializable)
+	===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
+	---! REMOVED METHOD: PUBLIC(-) io.opentelemetry.sdk.logs.SdkLogEmitterProvider getSdkLogEmitterProvider()
+	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.sdk.logs.SdkLoggerProvider getSdkLoggerProvider()
+***! MODIFIED CLASS: PUBLIC FINAL io.opentelemetry.sdk.OpenTelemetrySdkBuilder  (not serializable)
+	===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
+	---! REMOVED METHOD: PUBLIC(-) io.opentelemetry.sdk.OpenTelemetrySdkBuilder setLogEmitterProvider(io.opentelemetry.sdk.logs.SdkLogEmitterProvider)
+	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.sdk.OpenTelemetrySdkBuilder setLoggerProvider(io.opentelemetry.sdk.logs.SdkLoggerProvider)
```

Comment:
This is the most important part of the this change. **These are stable artifacts, and this is not a backwards compatible change.**

The `:sdk:all` module is stable, and has an `implementation` dependency on the unstable log SDK. We justified including references to the unstable the log SDK in the public API footprint by reasoning that for users to utilize the portions of the API related to logs, they must add a dependency on the log SDK (since its `implementation`, not `api`). In retrospect, we might have been better off with an internal method and reflection like we use to expose experimental features of the metrics SDK in [SdkMeterProviderUtil](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/SdkMeterProviderUtil.java#L31-L43). 

IMO, we made the decision with sound reasoning and are allowed to change the names of the methods / classes in these APIs. jApiCmp makes it painful to do so, but I've coerced it into allowing these changes. Maybe making it allowable but difficult to make these changes is the right thing.

Reply:
Clarified this as allowed behavior in #4729.